include/Gaffer/Context.inl

```diff
@@ -181,12 +181,16 @@ void Context::set( const IECore::InternedString &name, const T &value )
 	Storage &s = m_map[name];
 	if( Accessor<T>().set( s, value ) )
 	{
-		m_hashValid = false;
+		const std::string &nameStr = name.string();
+		s.hash = s.data->Object::hash();
+		s.hash.append( (uint64_t)&nameStr );
```

Comment:
I can try it ... I suspect it would be a measurable loss of performance?  I kinda wish there was a unique integer associated with InternedStrings based on insertion order, so that we could use that, and at least get the possibility of determinism for the tests if we pre-inserted the strings.  Though, actually, if we were going to do anything to InternedStrings to make them better for use in Gaffer, it would be tempting to store hashes with the central InternedString ... it seems pretty rare that we create a string in Gaffer and don't need to hash it at some point.

Reply:
Happy to merge as we are and then do the experiment after. Storing a hash with the InternedString itself sounds interesting - I wonder how much of the cost associated with hashing the string itself is MurmurHash and how much is just the memory access. Another experiment for after the merge perhaps?